tests/conftest.py

```diff
@@ -97,3 +99,31 @@ def fake_credentials() -> Credentials:
     )
     fake_credentials = with_scopes_if_required(fake_credentials, scopes=SCOPES)
     return fake_credentials
+
+
+def mock_server(server_sock: socket.socket) -> None:
+    """Create mock server listening on specified ip_address and port."""
+    ip_address = "127.0.0.1"
+    port = 3307
+    server_sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    server_sock.bind((ip_address, port))
+    server_sock.listen(0)
+    server_sock.accept()
```

Comment:
Will this only accept a single connection? 

Should we be checking to make sure that the connection is valid and we can send content through it?

Reply:
it would maybe be nice to test that the connection works, but not necessarily required.

Once thing is that it's fine for there to be overlap between unit and integration tests, because the unit tests are run locally and so should try to catch more issues than we need the integration tests to do.